src/haz3lcore/statics/MakeTerm.re

```diff
@@ -138,18 +138,18 @@ let rec go_s = (s: Sort.t, skel: Skel.t, seg: Segment.t): any =>
 and exp = unsorted => {
   let (term, inner_ids) = exp_term(unsorted);
   let ids = ids(unsorted) @ inner_ids;
-  return(e => Exp(e), ids, {ids, term});
+  return(e => Exp(e), ids, {ids, copied: false, term});
```

Comment:
I think the reason I left it like this is specifically because maketerm provides its own ids instead of letting fresh generate them. If you're going to make your own ids, I want you to pinky promise that the ids are unique by setting the copied flag. Come back at me if you disagree.